### DISCUSSION TITLE
Change Group - All Users Should Not Include Current User

### DISCUSSION THREAD

**Discussion by joeinnes** [directus/directus#5044] [Feature Requests]
Not really a bug, more of a consequence that might not be fully appreciated by users:

When adding users to a role through the Roles & Permissions dialogue (admin/settings/roles), it is possible to 'Select All' existing users to add them into the group. If the permissions are set restrictively on this role, then it can have the unintended consequence of adding the *current* user to the group, potentially locking the user out of the app.

I did this to myself and was able to fix it by resetting the role for my user, but the following would probably be a good idea:

1. IF the selected users includes the current user, warn and prompt for confirmation
2. IF completing this operation would leave 0 users with the 'Admin' role, then refuse to execute it.

**Comment by rijkvanzanten** [comment]
Generally speaking, I think a more fool-proof solution could be to prevent users from changing their own role at all times through a validation rule in app access permissions. Can we think of any reason why you would change your own role as a logged-in user?

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
